### ISSUE TITLE
"changed setup" not shown in context

### ISSUE THREAD

**Issue by testbird** [deltachat/deltachat-core#192]
There is no notification about messages from changed setups in group chats. The system message appears only in the user's chat.

**Comment by r10s** [comment]
that's true.
however, adding the "setup changed" message in every group may be lots of messages. not sure if this is wanted or needed.

**Comment by testbird** [comment]
I had expected the info about the changed setup to appear right together (or even within the message) with the changed setup (here, in the group chat). Why showing half of the information and sticking the other half elsewere. I was missing it, so I looked for it in the one-to-one chat, but what if there is no one-to-one chat with the user.

**Comment by hpk42** [comment]
Maybe a group message could have a special "!" icon or so when the sender
uses a different key and then you can go to info and see more details there?

**Comment by testbird** [comment]
It's a rare but important case. The current "tagline system message" format is fine, I think, just by prepending (inserting) that info text bubble into the message's text, instead of the user chat, the user gets the info *right where and in the form they are looking for*. And they get to know the affected messages and can ask the sender through other means.

**Comment by testbird** [comment]
The difference could maybe referred to as selecting a "message info" text vs. a "system message"?

**Comment by testbird** [comment]
Maybe the new "message info" used for displaying smtp errors can also be used here.

**Comment by testbird** [comment]
What do you think needs be "discussed" about deltachat showing status change warnings not within the message?

I had thoght you implemented inserting strings as "messege info" into the messages, after this bug was filed. That looked like the perfect place to provide status change warnings about a new message.

Having the warning in the "offending" message, and possibly the current status also visible in the contacts "avatar" (i.e. chat list, chat, and profile headers) would seem perfect to me.

**Comment by r10s** [comment]
> What do you think needs be "discussed" about deltachat showing status change warnings not within the message?

see above - there are different approaches and meanings. it's not that clear what to do, there are several paths one can follow.

**Comment by r10s** [comment]
> I would expect mismatching messages should never show up in these

if "mismatching" means changed keys - this is true.
in general, we have the rought idea to avoid too many "changed keys" warning as in signal or other messenger. users get used to it and nothing is won. verified groups is a step to this approach. showing "changed keys" messages in every group the user with the changed key is in _may_ be a step in the wrong direction.
all this may need more research and also user testing.

**Comment by testbird** [comment]
> users get used to [change notifications] and nothing is won

That's true. Therefore the "avatar" status indicator (something red for example, if contact has a verfied key).

> showing "changed keys" messages in every group the user with the changed key is in may be a step in the wrong direction.

Sure, but where comes this idea from? It should be enough to mark the individual mismatching message as such ("unverfied key" or something), it usually only appears in one (group) chat (possibly separate ad-hoc group if verfied groups are closely shielded).

**Comment by r10s** [comment]
> It should be enough to mark the individual mismatching message as such ("unverfied key" or something)

not sure. 
assume Alive and Bob are both in GroupG and GroupF.
Alice is sending a message with a new key to GroupG.
if Bob chats in GroupB meanwhile he won't be aware of the new key when answering.

**Comment by testbird** [comment]
Ok, now I understand your thoughts. Haven't thoght about that, yet.

**Comment by r10s** [comment]
another question that come into my mind when marking individual mismatching messages:
for how long would mismatching messages being marked?
or when is a new key "accepted" and no longer warned?

**Comment by testbird** [comment]
Ok, here are my conclusions, enhanced thanks to your posts:

1) Explaining the problem to the user is still missing a explicative and descriptive language. My idea, if not to talk about public (encryption) keys and private (decryption) keys would be to replace the term(s) "(deltachat) setup (message)", and instead talk about having and "Autocrypt ID".

* The users "Autocrypt ID" is confidential but may be tranfered (pin encrypted).
* The "Autocrypt ID" of contacts may be verfied (by checking its fingerprint).
...

2) Acknowledging that the contact verifcation status has a global status (within the app), but at the current autocrypt level it is not to be considered by default. When following this (insecure) default, status degradations must not bother the user by default, but must still be taken serious once a contact's Autocrypt ID got verified, and require user action to maintain the verification then. Therefore:

* Only show the degraded contact status info for contacts for contacts whose Autocrypt ID has been verified before.
* If a verfied contact's status degrades, show a warning in the global chat list.
* The actionable warning entries could be listed at the top of dc_get_chatlist()
* Providing the options to "re-verify" the contact's Autocrypt ID (e.g. fingerprint comparison over the phone), or to "ignore+remove the contact from all verfied chats".
* Within all the affected chats (normal or group chats) also show the same actionable warnings about any degraded members, instead of text entry field. (Requires action before a new message can be sent.)
* No "mismatching messages" should ever show up in verfied chats.
* Instead "mismatching messages" could open in an ad-hoc group (with the message containing the warning) until the Autocrypt ID that was used to send the message gets verified for that contact. (The ad-hoc chat could allow continuing the communication with the sender, or the whole group.)
* Not sure if this makes sense if trying to ensure responsible usage: The actionable warning (or some advanced setting) might allow to reset the Autocrypt ID of the once verified contact back to "unknown" (not checked).
